HDF5Examples/CMakeLists.txt

```diff
@@ -0,0 +1,170 @@
+cmake_minimum_required (VERSION 3.12)
+project (H5EXAMPLES C CXX)
+
+#-----------------------------------------------------------------------------
+# Basic HDF5Examples stuff here
+#-----------------------------------------------------------------------------
+include (${H5EXAMPLES_SOURCE_DIR}/config/cmake/HDFMacros.cmake)
+include (${H5EXAMPLES_SOURCE_DIR}/config/cmake/HDFExampleMacros.cmake)
+set (CMAKE_JAVA_INCLUDE_PATH "")
+
+SET_HDF_BUILD_TYPE()
+
+BASIC_SETTINGS (EXAMPLES)
+
+#-----------------------------------------------------------------------------
+# Define some CMake variables for use later in the project
+#-----------------------------------------------------------------------------
+set (HDF5EX_C_SRC_DIR          ${H5EXAMPLES_SOURCE_DIR}/C)
+set (HDF5EX_F90_SRC_DIR        ${H5EXAMPLES_SOURCE_DIR}/FORTRAN)
+set (HDF5EX_JAVA_DIR           ${H5EXAMPLES_SOURCE_DIR}/JAVA)
+
+#-----------------------------------------------------------------------------
+# HDF5 support
+#-----------------------------------------------------------------------------
+HDF5_SUPPORT (TRUE)
+message (STATUS "HDF5 link libs: ${H5EX_HDF5_LINK_LIBS}")
+APIVersion(${HDF5_VERSION} H5_LIBVER_DIR)
+message (STATUS "HDF5 H5_LIBVER_DIR: ${H5_LIBVER_DIR}")
+
+#-----------------------------------------------------------------------------
+# parse the full version number from H5public.h and include in H5_VERS_INFO
+#-----------------------------------------------------------------------------
+find_file (_h5public_h H5public.h ${H5EX_HDF5_INCLUDE_DIRS})
+file (READ ${_h5public_h} _h5public_h_contents)
+
+string (REGEX REPLACE ".*#define[ \t]+H5_VERS_MAJOR[ \t]+([0-9]*).*$"
+    "\\1" H5_VERS_MAJOR ${_h5public_h_contents})
+string (REGEX REPLACE ".*#define[ \t]+H5_VERS_MINOR[ \t]+([0-9]*).*$"
+    "\\1" H5_VERS_MINOR ${_h5public_h_contents})
+string (REGEX REPLACE ".*#define[ \t]+H5_VERS_RELEASE[ \t]+([0-9]*).*$"
+    "\\1" H5_VERS_RELEASE ${_h5public_h_contents})
+string (REGEX REPLACE ".*#define[ \t]+H5_VERS_SUBRELEASE[ \t]+\"([0-9A-Za-z._-]*)\".*$"
+    "\\1" H5_VERS_SUBRELEASE ${_h5public_h_contents})
+
+if (WIN32)
+  set(CMAKE_TEST_LIB_DIRECTORY "${HDF5_TOOLS_DIR}")
+else ()
+  set(CMAKE_TEST_LIB_DIRECTORY "${HDF5_LIBRARY_PATH}")
+endif ()
+
+#-----------------------------------------------------------------------------
+# Option to Enable MPI Parallel
+#-----------------------------------------------------------------------------
+set (CMAKE_MODULE_PATH ${H5EXAMPLES_RESOURCES_DIR} ${CMAKE_MODULE_PATH})
+option (HDF_ENABLE_PARALLEL "Enable parallel build (requires MPI)" OFF)
```

Comment:
Looks like this CMakeLists file needs to be updated to use the options that are already in the library (ENABLE_PARALLEL, ENABLE_THREADSAFE, BUILD_FORTRAN, BUILD_JAVA)

Reply:
So, this would be true if I didn't also need to provide a standalone copy in the binaries that could could be used to verify the install as well as provide workable examples.